src/main/java/com/marklogic/spark/reader/document/DocumentRowSchema.java

```diff
@@ -46,6 +51,61 @@ public static DocumentMetadataHandle makeDocumentMetadata(InternalRow row) {
         return metadata;
     }
 
+    public static void populateCollectionsColumn(Object[] row, DocumentMetadataHandle metadata) {
+        UTF8String[] collections = new UTF8String[metadata.getCollections().size()];
+        Iterator<String> iterator = metadata.getCollections().iterator();
+        for (int i = 0; i < collections.length; i++) {
+            collections[i] = UTF8String.fromString(iterator.next());
+        }
+        row[3] = ArrayData.toArrayData(collections);
+    }
+
+    public static void populatePermissionsColumn(Object[] row, DocumentMetadataHandle metadata) {
```

Comment:
Have we talked about putting this function (or something like it) in the Java client?

Reply:
Yes - there's a `parsePermissions` method on `DocumentMetadataHandle` now, but this code is specific to Spark because of the `Object row`.